packages/nodes-base/nodes/HttpRequest/V1/HttpRequestV1.node.ts

```diff
@@ -1026,20 +1026,13 @@ export class HttpRequestV1 implements INodeType {
 					}
 
 					newItem.json = returnItem;
-
-					newItem.binary![dataPropertyName] = await this.helpers.prepareBinaryData(
-						response!.body as Buffer,
-						fileName,
-					);
+					binaryData = response!.body;
 				} else {
 					newItem.json = items[itemIndex].json;
-
-					newItem.binary![dataPropertyName] = await this.helpers.prepareBinaryData(
-						response! as Buffer,
-						fileName,
-					);
+					binaryData = response;
 				}
 
+				newItem.binary![dataPropertyName] = await this.helpers.prepareBinaryData(binaryData);
```

Comment:
V1 of HTTP Request node does not return a fileName anymore.
Tried fetching https://upload.wikimedia.org/wikipedia/commons/3/3b/N8n-logo.png